### PR TITLE
refactor(operator): ensure build time parameter type safety for _subs…

### PR DESCRIPTION
**Pull request by kwonoj**
…cribe, call.

Try to close https://github.com/ReactiveX/RxJS/issues/291

### CODE REVIEW

File: src/operators/sample.ts [hunk 1]
```diff
@@ -12,8 +12,8 @@ class SampleOperator<T, R> implements Operator<T, R> {
   constructor(private notifier: Observable<any>) {
   }
   
-  call(observer: Observer<R>) {
-    return new SampleSubscriber(observer, this.notifier);
+  call(subscriber: Subscriber<R>) {
```
Comment (kwonoj) [inline src/operators/sample.ts]: Not sure missing return type for some cases (sample, sampletime, etcs) are intended or not. To limit scope of changes, did not touch.

Comment (blesh) [inline src/operators/sample.ts]: I think it's inferred by inheritance. It's probably fine.

**Comment by blesh** [comment]
Looks great! I'll be pulling this down to test it and merge soon.

**Comment by kwonoj** [comment]
Cool, thanks!

**Comment by blesh** [comment]
merged as of 88979a4

Thanks again @kwonoj!

**Pull request closed without merging**